src/Raven.Studio/typescript/models/database/index/indexDefinition.ts

```diff
@@ -122,20 +121,39 @@ class indexDefinition {
 
         this.reduce.extend({
             required: {
-                onlyIf: () => this.hasReduce()
-            }
+                onlyIf: () => this.hasReduce() && !this.reduce()
+            },
+            validation: [
+                {
+                    validator: (reduceContent: string) => {
+                        if (reduceContent && reduceContent.trim())
+                            return true; 
+                    },
+                    message: `Reduce function is empty!`
+                }
+            ]
         });
 
         this.reduceOutputCollectionName.extend({
             required: {
                 onlyIf: () => this.hasReduce() && this.outputReduceToCollection()
-            }
+            },
+            validation: [
+                {
+                    validator: () => !this.hasReduce() || this.reduce() || !this.outputReduceToCollection(),
+                    message: `Reduce function above is empty!`
```

Comment:
let's avoid word: above / below in validation messages.
Why in reduce function emptyness we check outputReduceToCollection?

Reply:
done - removed 'above'

Reply:
ok:)